src/write_outputs/long_duration_storage/write_opwrap_lds_stor_init.jl

```diff
@@ -32,4 +32,52 @@ function write_opwrap_lds_stor_init(path::AbstractString,
     CSV.write(joinpath(path, "StorageInit.csv"),
         dftranspose(dfStorageInit, false),
         header = false)
+
+    # Write storage evolution over full time horizon
+    hours_per_subperiod = inputs["hours_per_subperiod"];
+    t_interior = 2:hours_per_subperiod
+    T = hours_per_subperiod*NPeriods
+    SOC_t = zeros(G, T)
+    stor_long_duration = inputs["STOR_LONG_DURATION"]
+    stor_hydro_long_duration = inputs["STOR_HYDRO_LONG_DURATION"]
+    period_map = inputs["Period_Map"].Rep_Period_Index
+    pP_max = inputs["pP_Max"]
+    e_total_cap = value.(EP[:eTotalCap])
+    v_charge = value.(EP[:vCHARGE])*1000
+    v_P = value.(EP[:vP])*1000
+    if !isempty(stor_hydro_long_duration)
+        v_spill = value.(EP[:vSPILL])
+    end
+    for r in 1:NPeriods
+        w = period_map[r]
+        t_r = hours_per_subperiod * (r - 1) + 1
+        t_start_w = hours_per_subperiod * (w - 1) + 1
+        t_interior = 2:hours_per_subperiod
+
+        if !isempty(stor_long_duration)
+            SOC_t[stor_long_duration, t_r] = socw[stor_long_duration, r] .* (1 .- self_discharge.(gen[stor_long_duration])) .+ efficiency_up.(gen[stor_long_duration]) .* v_charge[stor_long_duration, t_start_w] .- 1 ./ efficiency_down.(gen[stor_long_duration]) .* v_P[stor_long_duration, t_start_w]
+
+            for t_int in t_interior
+                t = hours_per_subperiod * (w - 1) + t_int
+                SOC_t[stor_long_duration, t_r + t_int - 1] = SOC_t[stor_long_duration, t_r + t_int - 2] .* (1 .- self_discharge.(gen[stor_long_duration])) .+ efficiency_up.(gen[stor_long_duration]) .* v_charge[stor_long_duration, t] .- 1 ./ efficiency_down.(gen[stor_long_duration]) .* v_P[stor_long_duration, t]
+            end
+        end
+
+        if !isempty(stor_hydro_long_duration)
+            SOC_t[stor_hydro_long_duration, t_r] = socw[stor_hydro_long_duration, r] .- 1 ./ efficiency_down.(gen[stor_long_duration]) .* v_P[stor_hydro_long_duration, t_start_w] .- v_spill[stor_hydro_long_duration, t_start_w] .+ pP_max[stor_hydro_long_duration, t_start_w] .* e_total_cap[stor_hydro_long_duration]
+
+            for t_int in t_interior
+                t = hours_per_subperiod * (w - 1) + t_int
+                SOC_t[stor_hydro_long_duration, t_r + t_int - 1] = SOC_t[stor_hydro_long_duration, t_r + t_int - 2] .- 1 ./ efficiency_down.(gen[stor_long_duration]) .* v_P[stor_hydro_long_duration, t] .- v_spill[stor_hydro_long_duration, t] .+ pP_max[stor_hydro_long_duration, t] .* e_total_cap[stor_hydro_long_duration]
+            end
+        end
+    end
+    # println(SOC_t[10,:])
+    # error("STOP")
+    df_SOC_t = DataFrame(Resource = inputs["RESOURCE_NAMES"], Zone = zones)
+    df_SOC_t = hcat(df_SOC_t, DataFrame(SOC_t, :auto))
+    auxNew_Names = [Symbol("Resource"); Symbol("Zone"); [Symbol("n$t") for t in 1:T]]
```

Comment:
In the codebase, `T` typically refers to the number of time steps read from the `demand` file (which is stored in `inputs["T"]`). If I’m correct, the T at this line represents the length of the sub-periods multiplied by their count (and can therefore be different from `inputs["T"]`). Could you please use a different name to improve code clarity?